### PR TITLE
ecp-data-vis-sdk: Drop fortran from ascent spec.

### PR DESCRIPTION
**Pull request by kwryankrattiger**
@chuckatkins @cyrush

### CODE REVIEW

File: var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py [hunk 1]
```diff
@@ -125,7 +125,9 @@ def exclude_variants(variants, exclude):
     dav_sdk_depends_on('sensei@develop +vtkio +python ~miniapps', when='+sensei',
                        propagate=dict(propagate_to_sensei))
 
-    dav_sdk_depends_on('ascent+mpi+fortran+openmp+python+shared+vtkh+dray~test',
+    # Fortran support with ascent is problematic on some Cray platforms so the
+    # SDK is explicitly disabling it until the issues are resolved.
+    dav_sdk_depends_on('ascent+mpi~fortran+openmp+python+shared+vtkh+dray~test',
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py:130]: Haven't looked at `dav_sdk_depends_on` but, given the comment, would it be better to retain the existing dependency and add one that disables fortran `when='platform=cray +ascent'`?

Comment (kwryankrattiger) [reply var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py:130]: My understanding on this was that all platforms should provide the same spec if possible, but I will double check with @chuckatkins on that.

**Review by chuckatkins — APPROVED** [review]

File: var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py (hunk 1 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py:130]: If you are using the cray compiler wrappers, there is a way to skip deep cmake mpi detection.

We have a variant in our local spack package that supports this:
https://github.com/Alpine-DAV/ascent/blob/develop/scripts/uberenv_configs/packages/ascent/package.py#L68

Comment (cyrush) [reply var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py:130]: but that would also be a spec change.

Comment (kwryankrattiger) [reply var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py:130]: Ported with #30735 30735

**Pull request merged**